components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -6574,4 +6574,276 @@ private void validateAndUpdateURITemplates(API api, int tenantId) throws APIMana
+     * @param gatewayLabel Gateway label
+     * @return List of Gateway Policies
+     * @throws APIManagementException
+     */
+    @Override
+    public GatewayPolicyData getLightweightGatewayPolicyMappings(String organization,
+            String gatewayLabel) throws APIManagementException {
+        return apiMgtDAO.getPolicyMappingUUIDByGatewayLabel(gatewayLabel, organization);
+    }
+
+    /**
+     * Get a lightweight policy mapping data for a particular mapping ID.
+     * This will not include the policy definition as it is bulky.
+     *
+     * @param policyMappingUUID Policy mapping UUID
+     * @param tenantDomain      Tenant domain
+     * @return Gateway Policy Data
+     * @throws APIManagementException
+     */
+    @Override
+    public GatewayPolicyData getGatewayPolicyMappingDataByPolicyMappingId(String policyMappingUUID, String tenantDomain)
+            throws APIManagementException {
+        GatewayPolicyData gatewayPolicyData = apiMgtDAO.getGatewayPolicyMappingMetadataByPolicyMappingUUID(
+                policyMappingUUID);
+        if (gatewayPolicyData != null) {
+            gatewayPolicyData.setGatewayLabels(
+                    apiMgtDAO.getGatewayPolicyMappingDeploymentsByPolicyMappingId(policyMappingUUID, tenantDomain));
+            gatewayPolicyData.setGatewayPolicies(apiMgtDAO.getGatewayPoliciesOfPolicyMapping(policyMappingUUID));
+        }
+        return gatewayPolicyData;
+    }
+
+    /**
+     * Check whether a policy mapping deployment exists for a given policy mapping UUID.
+     *
+     * @param gatewayPolicyMappingId Policy mapping UUID
+     * @param tenantDomain           Tenant domain
+     * @return True if a policy mapping deployment exists
+     * @throws APIManagementException
+     */
+    @Override
+    public boolean isPolicyMappingDeploymentExists(String gatewayPolicyMappingId, String tenantDomain)
+            throws APIManagementException {
+        boolean isPolicyMappingDeploymentExists = false;
+        Set<String> gatewayLabels = apiMgtDAO.getGatewayPolicyMappingDeploymentsByPolicyMappingId(
+                gatewayPolicyMappingId, tenantDomain);
+        if (gatewayLabels.size() > 0) {
+            isPolicyMappingDeploymentExists = true;
+        }
+        return isPolicyMappingDeploymentExists;
+    }
+
+    @Override
+    public boolean hasExistingDeployments(String tenantDomain, String gatewayLabel) throws APIManagementException {
+        return !StringUtils.isBlank(
+                apiMgtDAO.getGatewayPolicyMappingByGatewayLabel(tenantDomain, gatewayLabel));
+    }
+
+    @Override
+    public boolean isPolicyMetadataExists(String gatewayPolicyMappingId)
+            throws APIManagementException {
+        GatewayPolicyData metadata = apiMgtDAO.getGatewayPolicyMappingMetadataByPolicyMappingUUID(
+                gatewayPolicyMappingId);
+        return metadata != null && Objects.equals(metadata.getPolicyMappingId(), gatewayPolicyMappingId);
```

Comment:
why are we doing a objects,equal here? Isnt it a string matching?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12201/commits/1136a795360198388c6b4bd97029c20b0f898832

Thanks and Regards